graphql/plugin/gqlidgen/code_generator.gotpl

```
@@ -21,6 +21,10 @@ func (r *{{$node.Name}}) ID() GqlID {
         {{- end }}
     //-----------------------------------------------------------------------------------------------
     {{- end }}
+    // It's safe to return the same ID since the node itself will resolve to nil and the ID never gets used.
+    if r == nil {
```

@radazen had to add this line but this should be ok, lmk what you think!

---

Ah, interesting. I ran into something similar with making the `viewerAdmire` field nullable (w.r.t. trace span logging), but I assumed that was a special case because we check for the GqlID interface and call the `ID()` method directly there. Where are you seeing nil IDs causing trouble?

---

Without the addition, I'd run into pointer dereference errors for queries like below, where if a caption was provided, it wants the event returned too.

```
# $input does not contain a caption

mutation useCreateCollectionMutation {
  createCollection(input: $input) {
    ... on CreateCollectionPayload {
      feedEvent {
        __typename
      }
    }
  }
}
```

Also happy to pair on this next week too

---

Gotcha. Yeah, let's follow up next week! I've been under the impression that gqlgen (and GraphQL in general) wouldn't attempt to resolve child fields on something that comes back `null`, but maybe it still attempts to resolve the ID?

My only concern with making GqlID _not_ throw an error is that we might silently ignore some real errors (referencing the ID of a thing that shouldn't exist, getting the empty string, treating it as a valid ID in the Relay cache or something, etc).

---

Sounds good!

---

I thought so too, seems like gqlgen chooses to always run extensions even if a field is null?

---

Ah! Are you specifically hitting the tracing error in the FieldReporter extension? Or is it somewhere else?

---

doh yes pulling in `main` fixed it!